### PR TITLE
Longhorn v1.1.1 release for branch "master"

### PR DESCRIPTION
**Pull request by cclhsu**
Longhorn v1.1.0 master

Signed-off-by: Clark Hsu <clark.hsu@suse.com>

### CODE REVIEW

**Review by shuo-wu — CHANGES_REQUESTED** [review]
1. It's better to split into 2 commits:
    1. Copy `latest` to `v1.1.0`
    2. Update `latest` with v1.1.1 Chart
2.  The directory name is wrong.

**Comment by shuo-wu** [comment]
The commit title doesn't match the actual content.

File: charts/longhorn/v1.1.0/Chart.yaml
Comment (jenting) [inline charts/longhorn/v1.1.0/Chart.yaml:4]: It should copy directly from the `lastest` folder (which is v1.1.0)

**Review by jenting — CHANGES_REQUESTED** [review]

**Comment by shuo-wu** [comment]
@cbron Hi, can you tell us what we need to do for the ci failure? I haven't seen this check previously.

**Comment by jenting** [comment]
> @cbron Hi, can you tell us what we need to do for the ci failure? I haven't seen this check previously.

The root cause is helm2 deprecated the URL https://kubernetes-charts.storage.googleapis.com.
The only way to fix is to use helm v2.17.0 in the CI test.

**Review by jenting — APPROVED** [review]

**Comment by shuo-wu** [comment]
The 1st commit is like renaming `latest` to `v1.1.0`

**Comment by shuo-wu** [comment]
> The 1st commit is like renaming `latest` to `v1.1.0`

^^ @cclhsu Check this comment

**Review by jenting — DISMISSED** [review]

**Review by shuo-wu — DISMISSED** [review]

**Comment by cclhsu** [comment]
@cbron @PennyScissors Can you help review this PR.  We would like to have the Longhorn Chart v1.1.1 release since Longhorn v1.1.1 is already released.  Thanks!

**Comment by innobead** [comment]
cc @khushboo-rancher

**Comment by PennyScissors** [comment]
The CI is currently broken due to a custom image that hasn't been updated, but you can do the following to test the chart yourself locally.

1. Add rancher-charts remote to your fork and fetch master
`git remote add -ft master rancher-charts https://github.com/rancher/charts.git`

2. Run ct lint
```
docker run -v $(pwd):/rancher-charts -w /rancher-charts quay.io/helmpack/chart-testing:latest ct lint \
--remote rancher-charts \
--target-branch master \
--validate-maintainers=false \
--chart-dirs charts/<chart-dir-name>
```

**Comment by cclhsu** [comment]
> The CI is currently broken due to a custom image that hasn't been updated, but you can do the following to test the chart yourself locally using the latest image.
> 
> 1. Add rancher-charts remote to your fork and fetch master
>    `git remote add -ft master rancher-charts https://github.com/rancher/charts.git`
> 2. Run ct lint
> 
> ```
> docker run -v $(pwd):/rancher-charts -w /rancher-charts quay.io/helmpack/chart-testing:latest ct lint \
> --remote rancher-charts \
> --target-branch master \
> --validate-maintainers=false \
> --chart-dirs charts/<chart-dir-name>
> ```

LINT pass against the PR branch.
```
cclhsu@dell-linux-nb:~/src/github.com/cclhsu_rancher/charts> REMOTE=rancher_charts
cclhsu@dell-linux-nb:~/src/github.com/cclhsu_rancher/charts> BRANCH=Longhorn_v1.1.0_master # master
cclhsu@dell-linux-nb:~/src/github.com/cclhsu_rancher/charts> CHART=longhorn
cclhsu@dell-linux-nb:~/src/github.com/cclhsu_rancher/charts> cd ~/src/github.com/cclhsu_rancher/charts
cclhsu@dell-linux-nb:~/src/github.com/cclhsu_rancher/charts> docker run -v $(pwd):/rancher-charts -w /rancher-charts quay.io/helmpack/chart-testing:latest ct lint \
> --target-branch ${BRANCH} \
> --validate-maintainers=false \
> --chart-dirs charts/${CHART}
Linting charts...
------------------------------------------------------------------------------------------------------------------------
No chart changes detected.
------------------------------------------------------------------------------------------------------------------------
All charts linted successfully
```

**Comment by khushboo-rancher** [comment]
Validated the changes with Rancher cluster manager app. The chart looks good.

**Comment by PennyScissors** [comment]
@cclhsu Looks like the path to --chart-dirs might be incorrect and it couldn't find any charts. The output should look like this.
```
docker run -v $(pwd):/rancher-charts -w /rancher-charts quay.io/helmpack/chart-testing:latest ct lint \
--remote rancher-charts \
--target-branch master \
--validate-maintainers=false \
--chart-dirs charts/longhorn  
Linting charts...

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 longhorn => (version: "1.1.1", path: "charts/longhorn/latest")
 longhorn => (version: "1.1.0", path: "charts/longhorn/v1.1.0")
------------------------------------------------------------------------------------------------------------------------

Linting chart 'longhorn => (version: "1.1.1", path: "charts/longhorn/latest")'
Checking chart 'longhorn => (version: "1.1.1", path: "charts/longhorn/latest")' for a version bump...
Old chart version: 1.1.0
New chart version: 1.1.1
Chart version ok.
Validating /rancher-charts/charts/longhorn/latest/Chart.yaml...
Validation success! 👍
charts/longhorn/latest/Chart.yaml
  28:113    error    no new line character at the end of file  (new-line-at-end-of-file)

Linting chart 'longhorn => (version: "1.1.0", path: "charts/longhorn/v1.1.0")'
Checking chart 'longhorn => (version: "1.1.0", path: "charts/longhorn/v1.1.0")' for a version bump...
Unable to find chart on master. New chart detected.
Validating /rancher-charts/charts/longhorn/v1.1.0/Chart.yaml...
Validation success! 👍
==> Linting charts/longhorn/v1.1.0

1 chart(s) linted, 0 chart(s) failed
Error: Error linting charts: Error processing charts
------------------------------------------------------------------------------------------------------------------------
 ✖︎ longhorn => (version: "1.1.1", path: "charts/longhorn/latest") > Error waiting for process: exit status 1
 ✔︎ longhorn => (version: "1.1.0", path: "charts/longhorn/v1.1.0")
------------------------------------------------------------------------------------------------------------------------
Error linting charts: Error processing charts
```

**Comment by meldafrawi** [comment]
LGTM

**Comment by cclhsu** [comment]
Fixed the end-of-line error
1 chart(s) linted, 0 chart(s) failed
------------------------------------------------------------------------------------------------------------------------
 ✔︎ longhorn => (version: "1.1.1", path: "charts/longhorn/latest")
 ✔︎ longhorn => (version: "1.1.0", path: "charts/longhorn/v1.1.0")
------------------------------------------------------------------------------------------------------------------------

**Review by PennyScissors — APPROVED** [review]

**Comment by cclhsu** [comment]
@cbron Please help review the PR.

**Review by cbron — APPROVED** [review]
I just skimmed, most changes are functional so I trust that they are right and QA has already validated. Once we merge this it will go live to production clusters so we must be sure it is good. So I will leave it to longhorn team to merge.

**Comment by innobead** [comment]
> I just skimmed, most changes are functional so I trust that they are right and QA has already validated. Once we merge this it will go live to production clusters so we must be sure it is good. So I will leave it to longhorn team to merge.

I don't have permission to do that, so probably need @yasker help on that.

**Comment by yasker** [comment]
Merged. Thanks @cbron @PennyScissors .

**Pull request merged**
